mne_bids/tests/test_mne_bids.py

```diff
@@ -242,60 +250,59 @@ def test_edf():
 
 
 def test_bdf():
-    """Test raw_to_bids conversion for Biosemi data."""
+    """Test write_raw_bids conversion for Biosemi data."""
     output_path = _TempDir()
     data_path = op.join(base_path, 'edf', 'tests', 'data')
     raw_fname = op.join(data_path, 'test.bdf')
 
-    raw_to_bids(subject_id=subject_id, session_id=session_id, run=run,
-                task=task, acquisition=acq, raw_file=raw_fname,
-                output_path=output_path, overwrite=False, kind='eeg')
+    bids_fname = partial_bids_fname + '_eeg.bdf'
+    raw = read_raw(raw_fname)
+    write_raw_bids(raw, bids_fname, output_path, overwrite=False)
 
     cmd = ['bids-validator', '--bep006', output_path]
     run_subprocess(cmd, shell=shell)
 
 
 def test_set():
-    """Test raw_to_bids conversion for EEGLAB data."""
+    """Test write_raw_bids conversion for EEGLAB data."""
     # standalone .set file
     output_path = _TempDir()
     data_path = op.join(testing.data_path(), 'EEGLAB')
     raw_fname = op.join(data_path, 'test_raw_onefile.set')
 
-    raw_to_bids(subject_id=subject_id, session_id=session_id, run=run,
-                task=task, acquisition=acq, raw_file=raw_fname,
-                output_path=output_path, overwrite=False, kind='eeg')
+    bids_fname = partial_bids_fname + '_eeg.set'
+    raw = read_raw(raw_fname)
+    raw._filenames = [raw_fname]
```

Comment:
The reason for removing the hack is that with the new API, `mne_bids` absolutely requires `raw.filenames` to exist as otherwise, it doesn't know where the original files are stored. I think this is a reasonable assumption to make since you know the path when reading in the file -- so if it's missing, that's more a bug upstream in MNE that should be fixed. It does mean `.set` files cannot be converted with MNE development version but I can live with that for a couple of months ...